setup.sh

```diff
@@ -131,6 +134,12 @@ function install_macos() {
         echo "-------------------------"
         /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
     fi
+    #Gets number after decimal in python version
+    version=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 
+    
+    if [[ $version -ge 9 ]]; then               #Checks if python version >= 3.9
```

Comment:
Are you sure this works as expected in all circumstances (assuming different terminals like bash / zsh / ... and the "soon" release of python 3.10)?

I would think the output of this will be a string of "9" or "10" - which i'm not sure bash will convert properly (correct me if i'm wrong - i never trusted bash types)

Reply:
It intentionally excludes the decimal and everything to the left. 10 > 9, but 3.10 ≯  3.9, even though python3.10 is newer.

You can test it out with this script

```
PYTHON="python3.10"
ten=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 
PYTHON="python3.9"
nine=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 
PYTHON="python3.8"
eight=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 
PYTHON="python3.8.9"
eight_nine=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 
PYTHON="python3.9.0"
nine_zero=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 
PYTHON="python3.10.1"
ten_one=$(egrep -o 3.\[0-9\]+ <<< $PYTHON | sed 's/3.//g' ) 

for v in $ten $nine $nine_zero $ten_one ; do

    if [[ $v -ge 9 ]]; then     
        echo "3.${v} works"
    else
        echo "3.${v} failed"
    fi
done

for v in $eight $eight_nine ; do

    if [[ $v -ge 9 ]]; then     
        echo "3.${v} failed"
    else
        echo "3.${v} works"
    fi
done
```

These are the results for `bash` `sh` `ksh` and `zsh`
3.8.9, 3.9.0, 3.10.1 get trimmed to 3.8, 3.9 ,3.10 in the `echo` but that only makes a different in this test and not in the `setup.sh` script

```
% bash test.sh 
3.10 works
3.9 works
3.9 works
3.10 works
3.8 works
3.8 works

% zsh test.sh 
3.10 works
3.9 works
3.9 works
3.10 works
3.8 works
3.8 works

% sh test.sh 
3.10 works
3.9 works
3.9 works
3.10 works
3.8 works
3.8 works

% ksh test.sh 
3.10 works
3.9 works
3.9 works
3.10 works
3.8 works
3.8 works
```